### PR TITLE
StringNonEmpty validator

### PR DESCRIPTION
**Pull request by LuizMoratelli**
string-non-empty-validator

### CODE REVIEW

**Comment by fakenickels** [comment]
hmm you did changes to just to include this new validator but there is a diff for the whole file, this makes the review harder. Can you investigate why the whole file is being changed? I thought it was a conversion from spaces to tabs but doesn't look like

File: packages/bs-reform/src/ReFormNext.re [hunk 1]
```diff
@@ -16,7 +16,7 @@ module Make = (Config: Config) => {
   module Validation = {
     type t =
       | Email(Config.field(string)): t
-      | Optional(Config.field('a)): t
+      | StringNonEmpty(Config.field(string)): t
```
Comment (fakenickels) [inline packages/bs-reform/src/ReFormNext.re]: actually Optional should stay! Optional is for fields that don't require any validation. Maybe we should rename it?

Comment (LuizMoratelli) [reply packages/bs-reform/src/ReFormNext.re]: I will keep Optional validator. But about the name, maybe it needs to be renamed, do you have any ideas?

Comment (yurijean) [reply packages/bs-reform/src/ReFormNext.re]: `NoValidation`

Comment (yurijean) [reply packages/bs-reform/src/ReFormNext.re]: or `Any`

Comment (fakenickels) [reply packages/bs-reform/src/ReFormNext.re]: NoValidation is nice

**Pull request merged**